intermediate_source/compiled_autograd_tutorial.rst

```diff
@@ -0,0 +1,301 @@
+* More prone to recompiles and graph breaks in dynamo due to the larger capture
+
+.. note:: Compiled Autograd is under active development and is not yet compatible with all existing PyTorch features. For the latest status on a particular feature, refer to `Compiled Autograd Landing Page <https://docs.google.com/document/d/11VucFBEewzqgkABIjebZIzMvrXr3BtcY1aGKpX61pJY>`_.
+
+Setup
+-----
+In this tutorial, we will base our examples on this simple neural network model.
+It takes a a 10-dimensional input vector, processes it through a single linear layer, and outputs another 10-dimensional vector.
+
+.. code:: python
+
+   import torch
+
+   class Model(torch.nn.Module):
+      def __init__(self):
+         super().__init__()
+         self.linear = torch.nn.Linear(10, 10)
+
+      def forward(self, x):
+         return self.linear(x)
+
+Basic usage
+------------
+Before calling the torch.compile API, make sure to set ``torch._dynamo.config.compiled_autograd`` to ``True``:
+
+.. code:: python
+
+   model = Model()
+   x = torch.randn(10)
+
+   torch._dynamo.config.compiled_autograd = True
+   @torch.compile
+   def train(model, x):
+      loss = model(x).sum()
+      loss.backward()
+
+   train(model, x) 
+
+In the code above, we create an instance of the ``Model`` class and generate a random 10-dimensional tensor ``x`` by using torch.randn(10).
+We define the training loop function ``train`` and decorate it with @torch.compile to optimize its execution.
+When ``train(model, x)`` is called:
+
+* Python Interpreter calls Dynamo, since this call was decorated with ``@torch.compile``
+* Dynamo intercepts the python bytecode, simulates their execution and records the operations into a graph
+* AOTDispatcher disables hooks and calls the autograd engine to compute gradients for ``model.linear.weight`` and ``model.linear.bias``, and records the operations into a graph. Using ``torch.autograd.Function``, AOTDispatcher rewrites the forward and backward implementation of ``train``.
+* Inductor generates a function corresponding to an optimized implementation of the AOTDispatcher forward and backward
+* Dynamo sets the optimized function to be evaluated next by Python Interpreter
+* Python Interpreter executes the optimized function, which basically executes ``loss = model(x).sum()``
+* Python Interpreter executes ``loss.backward()``, calling into the autograd engine, which routes to the Compiled Autograd engine since we enabled the config: ``torch._dynamo.config.compiled_autograd = True``
+* Compiled Autograd computes the gradients for ``model.linear.weight`` and ``model.linear.bias``, and records the operations into a graph, including any hooks it encounters. During this, it will record the backward previously rewritten by AOTDispatcher. Compiled Autograd then generates a new function which corresponds to a fully traced implementation of ``loss.backward()``, and executes it with ``torch.compile`` in inference mode
```

Comment:
```suggestion
* Compiled Autograd computes the gradients for ``model.linear.weight`` and ``model.linear.bias``, and records the operations into a graph, including any hooks it encounters. During this process, it will record the backward previously rewritten by AOTDispatcher. Compiled Autograd then generates a new function which corresponds to a fully-traced implementation of ``loss.backward()`` and executes it with ``torch.compile`` in inference mode.
```